pandas/_libs/tslibs/period.pyx

```diff
@@ -1537,6 +1545,7 @@ def extract_ordinals(ndarray values, freq) -> np.ndarray:
         raise TypeError("extract_ordinals values must be object-dtype")
 
     freqstr = Period._maybe_convert_freq(freq).freqstr
+    freqstr = freq_to_period_freqstr(freq.n, freq.name)
```

Comment:
nice! I wonder, do we even need the first one? as is, could we directly just do
```python
freqstr = freq_to_period_freqstr(freq.n, freq.name)
```
and remove
```
freqstr = Period._maybe_convert_freq(freq).freqstr
```

Reply:
Thanks for the comment. Yes, I agree, we don't need the line
```freqstr = Period._maybe_convert_freq(freq).freqstr```
I removed it.